src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -728,6 +726,15 @@ void ClusterTaskManager::FillResourceUsage(rpc::ResourcesData &data) {
       by_shape_entry->set_backlog_size(backlog_it->second);
     }
   }
+
+  // Check whether resources have been changed.
+  auto last_heartbeat_resources = gcs_client_->NodeResources().GetLastResourceUsage();
+  std::unordered_map<std::string, double> local_resource_map(data.resource_load().begin(),
+                                                             data.resource_load().end());
+  ResourceSet local_resource(local_resource_map);
+  if (!last_heartbeat_resources->GetLoadResources().IsEqual(local_resource)) {
+    data.set_resource_load_changed(true);
+  }
```

Comment:
Are there any unit tests covering this change?

Reply:
No actually, I'll add one.